src/ethereum/prague/fork.py

```diff
@@ -454,16 +472,19 @@ def check_transaction(
 
     if isinstance(tx, (BlobTransaction, SetCodeTransaction)):
         if not isinstance(tx.to, Address):
-            raise InvalidBlock
+            raise TransactionTypeContractCreationError(tx[0])
 
     if isinstance(tx, SetCodeTransaction):
         if not any(tx.authorizations):
-            raise InvalidBlock
+            raise EmptyAuthorizationListError("empty authorization list")
 
     if sender_account.nonce != tx.nonce:
-        raise InvalidBlock
+        if sender_account.nonce > Uint(tx.nonce):
+            raise NonceMismatchError("nonce too low")
+        else:
+            raise NonceMismatchError("nonce too high")
```

Comment:
How about just:

```python
        if sender_account.nonce > Uint(tx.nonce):
            raise NonceMismatchError("nonce too low")
        elif sender_account.nonce < Uint(tx.nonce):
            raise NonceMismatchError("nonce too high")
```

I'm not sure the nested if statement adds any readability.